app/main/views/reset_password.py

```diff
@@ -139,3 +140,54 @@ def update_password(token):
                                email_address=email_address,
                                form=form,
                                token=token), 400
+
+
+@main.route('/change-password', methods=["GET", "POST"])
+@login_required
+def change_password():
+    form = ChangeOldPasswordForm()
+    if current_user.role == 'supplier':
+        dashboard_url = url_for('external.supplier_dashboard')
+    elif current_user.role == "buyer":
+        dashboard_url = url_for('external.buyer_dashboard')
+    elif current_user.role.startswith('admin'):
+        dashboard_url = url_for('external.admin_dashboard')
+    else:
+        dashboard_url = url_for('external.index')
+
+    if request.method == 'POST':
+        if form.validate_on_submit():
+            # Make sure old password is valid
+            user_json = data_api_client.authenticate_user(
+                current_user.email_address,
+                form.old_password.data)
+            if user_json:
+                response = data_api_client.update_user_password(
+                    current_user.id,
+                    form.password.data,
```

Comment:
I know this is how we do it when logging in, but it feels a bit weird to send the password in plain text like this. Doesn't it risk dropping out in a log file someone where if the request fails for some reason?

Reply:
The api client catches any `HTTPError`s and returns `False` (hence just checking for 'response' below) without logging. I think we're probably OK?